src/main/kotlin/racingcar/domain/Race.kt

```diff
@@ -0,0 +1,12 @@
+package racingcar.domain
+
+class Race(
+    names: List<String>,
+) {
+    val cars: List<Car> = names.map { Car(name = it) }
```

Comment:
```suggestion
class Race(
    val cars: List<Car>
) {
```

초기화 파라미터를 사용하여 프로퍼티를 만드는 부분과, 프로퍼티를 바로 주입하는 부분의 차이는 객체 생성 관점에서 어떠한 장단점이 있을까요?

Reply:
프로퍼티에 바로 주입하면 코드가 간단해지는 장점이 있을거같고
초기화 파라미터로 초기화하면 객체 생성을 제가 원하는 방식으로만 생성을 제약할수 있을거 같습니다.

프로퍼티에서 초기화하면 시작부터 특정 차량의 이동수를 0이 아닌 다른 값으로 줄 수 있고 
이것은 객체를 유연하게 사용할수 있게 할 수 있겠지만
현재 자동차 경주는 출발점이 무조건 0이므로 위와 같이 생성 방식을 제한하는 것이 더 좋다고 생각했습니다.

Reply:
매우 잘 파악해주셨어요! 

저는 프로퍼티에서 초기화하면 테스트에 용이한 생성자를 만들 수가 있다고 생각해요. 테스트를 위한 생성자는 너무 테스트를 위해서 풀어놓는게 아닌가요? 하는 의견도 있겠지만, 테스트에 용이하게 생성자에 대한 룰을 풀어놓는 부분은 같이 일하는 동료 간의 약속으로 풀어나갈 수 있어보여요.

만약에 자동차에 대한 생성에 대해 꼭 name을 써야되겠다 하면, [정적 팩토리 메서드 패턴](https://velog.io/@lychee/%EC%9D%B4%ED%8E%99%ED%8B%B0%EB%B8%8C-%EC%9E%90%EB%B0%94-%EC%95%84%EC%9D%B4%ED%85%9C-1.-%EC%83%9D%EC%84%B1%EC%9E%90-%EB%8C%80%EC%8B%A0-%EC%A0%95%EC%A0%81-%ED%8C%A9%ED%84%B0%EB%A6%AC-%EB%A9%94%EC%84%9C%EB%93%9C%EB%A5%BC-%EA%B3%A0%EB%A0%A4%ED%95%98%EB%9D%BC)을 통하여 from(), of() 같은 정적 메서드를 만들어 정적 메서드 내에서 자동차를 만들고 Race를 만들어 볼 듯 하네요.

한번 정적 팩토리 메서드로 자동차 생성에 대한 부분을 개선해 볼까요?